cmd/skoap/main.go

```diff
@@ -0,0 +1,318 @@
+
+	tlsCertFlag = "tls-cert"
+	tlsKeyFlag  = "tls-key"
+
+	verboseFlag = "v"
+
+	experimentalUpgradeFlag = "experimental-upgrade"
+)
+
+const (
+	usageHeader = `
+skoap - Skipper based reverse proxy with authentication.
+
+Use the skoap proxy to verify authorization tokens before forwarding requests, and optionally check OAuth2 realms
+and scope or team membership. In addition to check incoming requests, optionally set basic authorization headers
+for outgoing requests.
+
+The command supports two modes:
+- single route mode: when a target address is specified, only a single route is used and the authorization
+  parameters (realm and scopes or teams) are specified as command line flags.
+- routes configuration: supports any number of routes with custom predicate and filter settings. The
+  authorization parameters are set in the routes file with the auth and authTeam filters.
+
+When used with eskip configuration files, it is possible to apply detailed augmentation of the requests and
+responses using Skipper rules.
+
+https://github.com/zalando/skipper
+
+`
+
+	addressUsage    = `network address that skoap should listen on`
+	etcdUrlsUsage   = "urls of nodes in an etcd cluster, storing route definitions"
+	etcdPrefixUsage = "path prefix for skipper related data in etcd"
+
+	targetAddressUsage = `when authenticating to a single network endpoint, set its address (without path) as
+the -target-address`
+
+	preserveHeaderUsage = `when forwarding requests, preserve the Authorization header in the outgoing request`
+
+	realmUsage = `when target address is used to specify the target endpoint, and the requests need to be
+authenticated against an OAuth2 realm, set the value of the realm with this flag. Note, that in case of a routes
+file is used, the realm can be set for each auth filter reference individually`
+
+	scopesUsage = `a comma separated list of the OAuth2 scopes to be checked in addition to the token validation
+and the realm check`
+
+	teamsUsage = `a comma separated list of the teams to be checked in addition to the token validation and the
+realm check`
+
+	auditUsage = `enable audit log in single route mode`
+
+	auditBodyUsage = `set the limit of the audit log body`
+
+	routesFileUsage = `alternatively to the target address, it is possible to use a full eskip route
+configuration, and specify the auth() and authTeam() filters for the routes individually. See also:
+https://godoc.org/github.com/zalando/skipper/eskip`
+
+	insecureUsage = `when this flag set, skipper will skip TLS verification`
+
+	authUrlBaseUsage = `URL base of the authentication service. The authentication token found
+in the incoming requests will be validated agains this service. It will be passed as the Authorization Bearer
+header`
+
+	teamUrlBaseUsage = `URL base of the team service. The user id received from the authentication service will
+be appended to this url, and the list of teams that the user is a member of will be requested`
```

Comment:
An example would be good here.
Example URL: https://my-group-svc.example.org/:user , where :user will be replaced by the user got from the authentication service, for example jdoe.